Minesweeper.py

```diff
@@ -44,132 +70,143 @@ def __init__(self):
-        self.tiles = None
-        self.game_over = False
-        self.game_won = False
-        self.game_won_message_printed = False
-        # DHB: for a state variable like this, it's especially important to ensure that the valid states are well defined. You could for example use a game state
-        # DHB: enumeration class, similar to the color class, to define the valid status
-        self.state = "tutorial"
-        self.clock = pygame.time.Clock()
+
+        # Create the game window
+        self.win = pygame.display.set_mode((888, 938))
+
+        # sets how much time has passed
         self.time_elapsed = 0
+
+        # sets the time you start with
         self.start_time = 0
 
-        self.tutorial_screen = TutorialScreen()
-        self.game_won_screen = GameWonScreen()
-        self.reset_button = ResetButton()
+        # sets the cursor as visible
+        pygame.mouse.set_visible(True)
 
-    def add_game_objects(self, *list_of_game_objects, **dict_of_game_objects):
-        for game_object in list_of_game_objects:
-            self.add_game_object(game_object)
-        for game_object_id, game_object in dict_of_game_objects.items():
-            self.add_game_object(game_object)
+        # sets the look of the cursor
+        pygame.mouse.set_cursor(*pygame.cursors.tri_left)
 
-    def add_game_object(self, game_object):
-        self.game_objects.append(game_object)
-        if isinstance(game_object, Tiles):
-            self.tiles = game_object
+        self.tutorial_screen = TutorialScreen(self)
+        self.game_won_screen = GameWonScreen(self)
+        self.reset_button = ResetButton(self)
+        self.gamestate = GameState()
+        self.tiles = Tiles(self)
 
-    def get_game_objects_with_attribute(self, attribute_name):
-        return [game_object for game_object in self.game_objects if hasattr(game_object, attribute_name)]
+    def draw(self):
+        """calls the draw functions required under certain circumstances
+        """
+        events = pygame.event.get()
+        self.win.fill(Color.BACKGROUND)
+        self.tiles.draw()
+        self.reset_button.draw()
 
-    def draw(self): # calls all of the draw functions
-        
-        if self.game_over:
-            self.tiles.draw()
+        if self.gamestate.game_won:
+            self.win.fill(Color.BACKGROUND)
+            self.game_won_screen.draw()
 
-        # DHB: I wonder if this 'game_won_message_printed' flag is a good idea. Draw loops usually draw everything
-        # DHB: in every frame, but I imagine this will only make the message be drawn in one frame 
-        if self.game_won and not self.game_won_message_printed:
+        if self.gamestate.game_won:
             self.game_won_screen.draw()
-            self.game_won_message_printed = True
 
-        self.reset_button.draw()
+        if self.gamestate.game_over == True or self.gamestate.game_won == True:
+            self.game_won_screen.handle_events(events)
```

Comment:
handle_events should not be part of the draw function. have the main loop call both draw and handle_events functions on Game, and have the handle_events function call self.game_won_screen.handle_events.

Its very important in programming to try to keep concepts clear and separate concerns.